### PR TITLE
Article Block

### PR DESCRIPTION
**Pull request by hunyoboy**

### CODE REVIEW

File: src/ArticleDocumentMarshaler.php [hunk 1]
```diff
@@ -437,7 +441,47 @@ protected function afterTransformBlock(Message $block, array &$context): void
...
+        $stackDisplay = new HorizontalStackDisplay();
+
+        $container
+            ->setContentDisplay($stackDisplay)
+            ->setIdentifier($block->get('etag') . $context['idx'])
+            ->addAddition($linkAddition);
+
+        /** @var NodeRef $imageRef */
+        $imageRef = $block->get('poster_image_ref')
+            ?: $node->get('poster_image_ref')
```
Comment (bhsiao55) [inline src/ArticleDocumentMarshaler.php]: the node in this case is always an article.  article node doesn't have poster_image_ref

File: src/ArticleDocumentMarshaler.php [hunk 2]
```diff
@@ -437,7 +441,47 @@ protected function afterTransformBlock(Message $block, array &$context): void
...
+        $container = new Container();
+        $stackDisplay = new HorizontalStackDisplay();
+
+        $container
+            ->setContentDisplay($stackDisplay)
+            ->setIdentifier($block->get('etag') . $context['idx'])
+            ->addAddition($linkAddition);
+
+        /** @var NodeRef $imageRef */
+        $imageRef = $block->get('poster_image_ref')
```
Comment (bhsiao55) [inline src/ArticleDocumentMarshaler.php]: article block doesn't have poster_image_ref only image_ref

File: src/ArticleDocumentMarshaler.php [hunk 3]
```diff
@@ -437,7 +441,47 @@ protected function afterTransformBlock(Message $block, array &$context): void
...
+            ?: $node->get('poster_image_ref')
+                ?: $node->get('image_ref');
+
+        $imageUrl = $this->getImageUrl($imageRef, AspectRatio::R4BY3(), $block);
```
Comment (bhsiao55) [inline src/ArticleDocumentMarshaler.php]: lets use 1x1

File: src/ArticleDocumentMarshaler.php [hunk 4]
```diff
@@ -437,7 +441,47 @@ protected function afterTransformBlock(Message $block, array &$context): void
...
+
+        /** @var NodeRef $nodeRef */
+        $nodeRef = $block->get('node_ref');
+        $node = $this->getNode($nodeRef);
+        if (null === $node) {
+            return null;
+        }
+
+        $linkAddition = new ComponentLink();
+        $linkAddition->setURL($this->getCanonicalUrl($node));
```
Comment (bhsiao55) [inline src/ArticleDocumentMarshaler.php]: should we link to web version of article or try to keep it within apple news?
only text components can link to outside i believe containers can only access internal urls

File: src/ArticleDocumentMarshaler.php [hunk 5]
```diff
@@ -437,7 +441,47 @@ protected function afterTransformBlock(Message $block, array &$context): void
...
+
+        $imageUrl = $this->getImageUrl($imageRef, AspectRatio::R4BY3(), $block);
+
+        $image = new Image();
+        $image->setUrl($imageUrl);
+
+        $title = new Title();
+        $title->setText($node->get('title'));
+
+        $section = new Section();
```
Comment (bhsiao55) [inline src/ArticleDocumentMarshaler.php]: how about using aside https://developer.apple.com/documentation/apple_news/aside

**Comment by hunyoboy** [comment]
@bhsiao55 Great suggestion. Unit test added. Thanks.

**Comment by arunkarnati** [comment]
cleaned up transformArticleBlock code, I will create a new pull request with those changes.

**Pull request closed without merging**